packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -7,19 +7,25 @@ import {
   ResourceSnapshotsReportType,
   SnapshotExportContentFormat,
   ApplyOptionsDeletionScope,
+  SnapshotDiffModel,
 } from '@coveord/platform-client';
 import retry from 'async-retry';
 import {ReportViewer} from './reportPreviewer/reportPreviewer';
 import {ensureFileSync, writeJsonSync} from 'fs-extra';
 import {join} from 'path';
 import {SnapshotReporter} from './snapshotReporter';
 import {SnapshotOperationTimeoutError} from '../errors';
-import {ExpandedPreviewer} from './expandedPreviewer/expandedPreviewer';
 import {Project} from '../project/project';
 import {SnapshotNoReportFoundError} from '../errors/snapshotErrors';
-import {SnapshotReportStatus} from './reportPreviewer/reportPreviewerDataModels';
+import {SnapshotDiffReporter} from './diffReporter/diffReporter';
+import {CliUx} from '@oclif/core';
 
-export type SnapshotReport = ResourceSnapshotsReportModel;
+export type SnapshotReport = ResourceSnapshotsReportModel | SnapshotDiffModel;
+
+type SnapshotReportTypes = keyof Pick<
+  ResourceSnapshotsModel,
+  'reports' | 'diffGenerationReports'
+>;
```

Comment:
```suggestion
type SnapshotReportTypes = 'reports' | 'diffGenerationReports';
```

Or I missed something?

Reply:
No that's good. I used `Pick` so if there is a change in the interface, I would get an error in the type instead of the code.
But, I agree, it's complex for nothing.